pkgs/development/python-modules/synologydsm-api/default.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, poetry
+, requests
+, urllib3
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "synologydsm-api";
+  version = "1.0.2";
+
+  format = "pyproject";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "hacf-fr";
+    repo = "synologydsm-api";
+    rev = "v${version}";
+    sha256 = "0gyahf1x6i6j9pslh1y3pyh3si5jvxb06r1w761b9gsxyk14y1si";
+  };
+
+  nativeBuildInputs = [
+    poetry
```

Comment:
Should be migrated to poetry-core upstream, can you open an issue to that end?

https://github.com/NixOS/nixpkgs/issues/103325

Reply:
I'll file an upstream PR and apply that using `fetchpatch`.